indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/SimpleResourceManagementStrategy.java

```diff
@@ -174,16 +179,18 @@ public String apply(ZkWorker input)
       currentlyTerminating.clear();
       currentlyTerminating.addAll(stillExisting);
 
-      updateTargetWorkerCount(workerConfig, pendingTasks, zkWorkers);
+      Collection<ZkWorker> workers = runner.getWorkers();
+      updateTargetWorkerCount(workerConfig, pendingTasks, workers);
 
-      final Predicate<ZkWorker> isLazyWorker = createLazyWorkerPredicate(config);
       if (currentlyTerminating.isEmpty()) {
-        final int excessWorkers = (zkWorkers.size() + currentlyProvisioning.size()) - targetWorkerCount;
+
+        final int excessWorkers = (workers.size() + currentlyProvisioning.size()) - targetWorkerCount;
         if (excessWorkers > 0) {
+          final Predicate<ZkWorker> isLazyWorker = createLazyWorkerPredicate(config);
           final List<String> laziestWorkerIps =
-              FluentIterable.from(zkWorkers)
-                            .filter(isLazyWorker)
-                            .limit(excessWorkers)
+              FluentIterable.from(
```

Comment:
Since we are not limiting any more can we simplify this part? maybe just use either Lists.transform or Iterables.transform ?